Add stories and argstable for basic datepickers

b12f opened the pull request:
Closes https://community.openproject.org/work_packages/46645/activity

Commit:
Add stories and argstable for basic datepickers

Closes https://community.openproject.org/work_packages/46645/activity

Commit:
Merge branch 'dev' into storybook/46645-datepicker-examples

Commit:
Revert storybook-angular to v6

Commit:
Fix frontend module dependencies

Commit:
Export datepickers from shared module

Commit:
Merge branch 'dev' into storybook/46645-datepicker-examples

Kharonus commented on frontend/src/app/core/days/weekday.service.shim.ts:
I somehow have the feeling, that the shims are more stubs then shims. :D In addition, I see no benefit to add them right next to the service class they are replacing. Why not putting them to the stories directory? maybe in a subfolder there called `stubs` or `service-stubs`.

b12f replied:
I put them there because they might also be interesting for usage in TypeScript specs. I agree with the naming, I'll change that.

Kharonus reviewed (changes requested):
I think, the stories look quite fine. I'd just challenge the current form of the service stubs.

Commit:
Rename shims to stubs

wielinde commented:
Hey @b12f I just checked this branch out. For whatever reason the days of the weeks do not get rendered for any of the three stories:
![image](https://user-images.githubusercontent.com/327272/231807949-00d4a7e1-fa99-4f02-a4ce-843c21ff779b.png)

b12f commented:
> Hey @b12f I just checked this branch out. For whatever reason the days of the weeks do not get rendered for any of the three stories.

Ah! I think this is because I'm faking the translation files instead of loading them all: https://github.com/opf/openproject/pull/12376/files#diff-5c8447512de6ab867c557a8426da0992dfb66c183d6b3ec8d05428b3d90964beR4

I'll add those translations.

Commit:
Add day of the week names to i18n stub

psatyal committed:
Removed other date picker stories

Included relevant information from "Common elements" to the now-consolidated "Date picker" story.

psatyal commented:
Removed the other stories and extended the Basic Date Picker story to include relevant bits from the (now removed) _Common elements_ page.

psatyal commented:
@b12f, just checked out the pullpreview and I think the date picker docs do indeed make more sense like this. A few thoughts:

- For the building blocks, we could perhaps try embedding Figma frames directly for the examples?
- The Mini day section has a lot of specifications around colour. Do we think that this level of detail is appropriate for the DS docs?

Cheers
Parimal

cc @wielinde just to keep you posted.

psatyal committed:
Testing adding Figma iframe embeds

Commit:
Merge branch 'dev' into storybook/46645-datepicker-examples

Commit:
Add mdx-embed and a Figma example for embedding

Commit:
Merge branch 'dev' into storybook/46645-datepicker-examples

psatyal committed:
Added Figma examples

Commit:
Merge branch 'dev' into storybook/46645-datepicker-examples

Commit:
Fix modal banners

Commit:
Merge branch 'storybook/46645-datepicker-examples' of github.com:opf/openproject into storybook/46645-datepicker-examples

Commit:
Fix day service stub with new resource stores

Commit:
Fix linting errors

Commit:
Merge branch 'dev' into storybook/46645-datepicker-examples

Commit:
Merge branch 'dev' into storybook/46645-datepicker-examples

Kharonus reviewed (approved):
LGTM from code perspective
